broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStoragePartitionStep.java

```diff
@@ -90,4 +117,15 @@ private IllegalStateException buildWrongTermException(
         String.format(
             WRONG_TERM_ERROR_MSG, context.getCurrentTerm(), raftTerm, context.getPartitionId()));
   }
+
+  private static class LogAppenderForReadOnlyStorage implements ZeebeLogAppender {
+    @Override
+    public void appendEntry(
+        final long lowestPosition,
+        final long highestPosition,
+        final ByteBuffer data,
+        final AppendListener appendListener) {
+      throw new UnsupportedOperationException("Read only storage");
```

Comment:
Maybe we can give this exception a bit more context

Reply:
Lets follow our error guidelines https://github.com/camunda-cloud/zeebe/wiki/Error-Guidelines 

Like: "Expect to append entry {} on partition {}, but was in Follower role. Followers must not append entries to the log storage."